external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -95,8 +95,10 @@ private[kafka010] class KafkaOffsetReader(
    * Closes the connection to Kafka, and cleans up state.
    */
   def close(): Unit = {
-    consumer.close()
-    kafkaReaderThread.shutdownNow()
+    runUninterruptibly {
```

Comment:
If the kafkaReaderThread thread is using the consumer, `consumer.close` will throw `ConcurrentModificationException`. Put it inside `runUninterruptibly` to prevent this case happening.